console-subscriber/tests/support/mod.rs

```diff
@@ -62,3 +62,45 @@ where
         .spawn(f)
         .expect(&format!("spawning task '{name}' failed"))
 }
+
+/// Wakes itself from within this task.
+///
+/// This function returns a future which will wake itself and then
+/// return `Poll::Pending` the first time it is called. The next time
+/// it will return `Poll::Ready`.
+///
+/// This is the old behavior of Tokio's [`yield_now()`] function, before it
+/// was improved in [tokio-rs/tokio#5223] to avoid starving the resource
+/// drivers.
+///
+/// Awaiting the future returned from this function will result in a
+/// self-wake being recorded.
+///
+/// [`yield_now()`]: fn@tokio::task::yield_now
+/// [tokio-rs/tokio#5223]: https://github.com/tokio-rs/tokio/pull/5223
+#[allow(dead_code)]
+pub(crate) fn self_wake() -> impl Future<Output = ()> {
```

Comment:
Maybe we should move it to a separate crate. Then we don't need to duplicate it in the examples. But I think it is OK to have this kind of small duplication.

Reply:
I thought about that, but I didn't want to introduce additional complexity for the example script - also, this means that it can be copy-pasted elsewhere without non-published dependencies.